tools/pvp.py

```diff
@@ -2,32 +2,54 @@
 import sys
 import os
 import itertools
+import requests
 
 
 class PVP:
     def __init__(self, pa_root):
-        with open(pa_root + "/data/base_stats.json", "r") as json_data:
-            stats = json.load(json_data)
-            json_data.close()
+        # Fetch pokemon data
+        master_file = "https://raw.githubusercontent.com/WatWowMap/" \
+            "Masterfile-Generator/master/master-latest-everything.json"
+        master_file = requests.get(master_file)
+        monster_data = master_file.json()["pokemon"]
+        with open(pa_root + "/data/pokemon_data.json", 'w') as f:
```

Comment:
Also, why is this happening in pvp, this should probably have its own utility file that is loaded on PA start.

Reply:
> Ok, I see we are caching the contents of the remote file locally, but there is no error handling here, IE, if the remote file is missing or becomes empty how do we prevent it from overwriting the local cached copy?

The data will be erased only if the pokemon_data is missing or the local hash `data/.data_version` is different than the hash of the remote file. There is no reason why the file could be empty. If this is about connection issues, PA couldn't start sending the alarms anyway even with the local copy.


> Also, why is this happening in pvp, this should probably have its own utility file that is loaded on PA start.

`pvp.py` is an independent tool, which means this is not used by PA and doesn't include any of its modules to work. You could run `python3 tools/pvp.py` if you have the local `pokemon_data.json`.

It is used to calculate PvP stat products of Pokemons by getting their base stats. Before this PR, it used the `base_stats.json` file always available in `data/` folder locally. With this PR who fetch `pokemon_data.json` remotely, there is no way to know if the data is available in local `data/` folder when you run `python3 tools/pvp.py` as it won't start PA. 

Same reason for including [the download of the pokemon data when running the unit tests](https://github.com/pokealarm/pokealarm/pull/802/files#diff-e314724a0aa4e95190e5c3a2443ee29c11ff49771675ed554fd46ba14176a275R14).

Reply:
> The data will be erased only if the pokemon_data is missing or the local hash data/.data_version is different than the hash of the remote file. There is no reason why the file could be empty. If this is about connection issues, PA couldn't start sending the alarms anyway even with the local copy.

Let's suppose a rogue commit wipes or deletes the remote file. How are we going to deal with that dependency. The hash will be different but the remote file is no longer trustworthy. We need a way to fall-back on previously cached data so that PA's dependency is not impacted.